src/lib/api/ApiTransformers.ts

```diff
@@ -0,0 +1,390 @@
+		archived: channel.archived ?? false,
+		archivedTimestamp: channel.archiveTimestamp,
+		createdTimestamp: channel.createdTimestamp ?? 0,
+		guildId: channel.guildId,
+		name: channel.name,
+		parentId: channel.parentId,
+		permissionOverwrites: [...(channel.parent?.permissionOverwrites.cache.entries() ?? [])],
+		rawPosition: channel.parent?.rawPosition ?? null,
+		rateLimitPerUser: channel.rateLimitPerUser,
+	};
+}
+
+function flattenChannelFallback(channel: Channel | GuildChannel | DMChannel | ThreadChannel): FlattenedChannel {
+	return {
+		id: channel.id,
+		type: channel.type as FlattenedChannel['type'],
+		createdTimestamp: channel.createdTimestamp ?? 0,
+	};
+}
+
+export interface FlattenedChannel {
+	createdTimestamp: number;
+
+	id: string;
+
+	type: ChannelTypeString;
+}
+
+export interface FlattenedGuildChannel extends FlattenedChannel {
+	guildId: string;
+
+	name: string;
+
+	parentId: string | null;
+
+	permissionOverwrites: [string, PermissionOverwrites][];
+
+	rawPosition: number;
+
+	type: ChannelTypeString;
+}
+
+export interface FlattenedNewsChannel extends FlattenedGuildChannel {
+	type: ChannelType.GuildAnnouncement;
+
+	nsfw: boolean;
+
+	topic: string | null;
+}
+
+export interface FlattenedTextChannel extends FlattenedGuildChannel {
+	nsfw: boolean;
+
+	rateLimitPerUser: number;
+
+	topic: string | null;
+
+	type: ChannelType.GuildText;
+}
+
+export interface FlattenedThreadChannel extends Pick<FlattenedGuildChannel, 'id' | 'createdTimestamp'> {
+	archived: boolean;
+
+	archivedTimestamp: number | null;
+
+	guildId: string;
+
+	name: string;
+
+	parentId: string | null;
+
+	permissionOverwrites: [string, PermissionOverwrites][];
+
+	rateLimitPerUser: number | null;
+
+	rawPosition: number | null;
+
+	type: ChannelType.GuildPublicThread | ChannelType.GuildPrivateThread | ChannelType.GuildNewsThread;
+}
+
+export interface FlattenedNewsThreadChannel extends FlattenedChannel {
+	type: ChannelType.AnnouncementThread;
+}
+
+export interface FlattenedPublicThreadChannel extends FlattenedChannel {
+	type: ChannelType.PublicThread;
+}
+
+export interface FlattenedPrivateThreadChannel extends FlattenedChannel {
+	type: ChannelType.PrivateThread;
+}
+
+export interface FlattenedVoiceChannel extends FlattenedGuildChannel {
+	type: ChannelType.GuildVoice;
+
+	bitrate: number;
+
+	userLimit: number;
+}
+
+export interface FlattenedDMChannel extends FlattenedChannel {
+	type: ChannelType.DM;
+
+	recipient: string;
+}
+
+// #endregion Channel
+
+// #region User
+
+export function flattenUser(user: User): FlattenedUser {
+	return {
+		id: user.id,
+		bot: user.bot,
+		username: user.username,
+		discriminator: user.discriminator,
+		avatar: user.avatar,
+	};
+}
+
+export interface FlattenedUser {
+	avatar: string | null;
+
+	bot: boolean;
+
+	discriminator: string;
+
+	id: string;
+
+	username: string;
+}
+
+// #endregion User
+
+// #region Member
+
+export function flattenMember(member: GuildMember): FlattenedMember {
```

Comment:
Can you elaborate what the flatten Objects are for? A type with less information in it?

Reply:
We don't need to send an entire class as that would risk sending too much data for nothing.